pkg/common-controller/snapshot_controller.go

```diff
@@ -135,6 +135,13 @@ func (ctrl *csiSnapshotCommonController) syncContent(content *crdv1.VolumeSnapsh
 			// right away so that it is in-sync with the content status
 			ctrl.snapshotQueue.Add(snapshotName)
 		}
+
+		// Check if we need to remove the content Finalizer from the snapshot because the deletionPolicy has changed
+		if snapshot != nil && utils.NeedToRemoveSnapshotBoundFinalizer(snapshot, content) {
+			// We have the volumesnapshot bound finalizer, but our deletionPolicy has changed to Retain
+			klog.V(5).Infof("syncContent [%s]: Removing Finalizer for VolumeSnapshot", snapshot.Name)
+			return ctrl.removeContentFinalizer(snapshot)
```

Comment:
I'm not convinced that we need to remove utils.VolumeSnapshotBoundFinalizer from VolumeSnapshot in syncContent.  We've already got this logic in syncSnapshot.

Reply:
Looks like you're correct - I was testing with an older version of the controller, and it appears that this has been resolved as of v2.1.1.

Apologies. I'll close this PR.